### PR TITLE
refactor events utils in e2e

### PR DESCRIPTION
**Pull request by danielqsj**
**What type of PR is this?**

/kind feature
/sig testing

**What this PR does / why we need it**:

`test/e2e/common` is a mixture of tests (`test/e2e/common/downward_api.go`) and utility code (`test/e2e/common/events.go`). It would be nice to refactor it so that the utility code can be imported without also importing the tests. Most likely a better place for the utility code is under `test/e2e/framework`.

For this PR, the specific reason was the usage of `WaitTimeoutForEvent` in a storage test, which then made it impossible to import the storage tests without those other tests. See #83609. Once `WaitTimeoutForEvent` is in an independent utility package, the copied waitTimeoutForEvent can be removed from `test/e2e/storage/testsuites/volumemode.go` again.

**Which issue(s) this PR fixes**:

part of https://github.com/kubernetes/kubernetes/issues/83937

**Special notes for your reviewer**:

/cc @pohly

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: test/e2e/storage/testsuites/volumemode.go [hunk 1]
```diff
@@ -482,16 +482,6 @@ func listPodDirectory(h utils.HostExec, path string, node *v1.Node) ([]string, e
...
 }
 
-// NOTE(avalluri): The below code is intentionally copied from e2e/common package.
-// Testsuites depending on common package is not desirable as that pulls quite
-// many tests which are not interested by storage suites.
-//
-// waitTimeoutForEvent waits the given timeout duration for an event to occur.
-func waitTimeoutForEvent(c clientset.Interface, namespace, eventSelector, msg string, timeout time.Duration) error {
-	interval := 2 * time.Second
-	return wait.PollImmediate(interval, timeout, eventOccurred(c, namespace, eventSelector, msg))
```
Comment (pohly) [inline test/e2e/storage/testsuites/volumemode.go]: I think the `eventOccurred` below can now also be removed.

Comment (danielqsj) [reply test/e2e/storage/testsuites/volumemode.go]: Yep, I miss it. Thanks.
PR updated, PTAL

File: test/e2e/storage/testsuites/volumemode.go [hunk 2]
```diff
@@ -481,32 +480,3 @@ func listPodDirectory(h utils.HostExec, path string, node *v1.Node) ([]string, e
 	}
 	return strings.Split(out, "\n"), nil
 }
-
-// NOTE(avalluri): The below code is intentionally copied from e2e/common package.
```
Comment (andrewsykim) [inline test/e2e/storage/testsuites/volumemode.go]: Does this still apply?

Comment (pohly) [reply test/e2e/storage/testsuites/volumemode.go]: No, all of the copied code gets removed by this PR.

File: test/e2e/framework/events/events.go [hunk 3]
```diff
@@ -35,11 +35,12 @@ import (
 	"github.com/onsi/ginkgo"
```
Comment (oomichi) [inline test/e2e/framework/events/events.go]: The dependencies seem clean and good for me for moving the package into e2e framework.

**Review by oomichi — COMMENTED** [review]
Thanks for doing this.

/lgtm

**Comment by pohly** [comment]
@danielqsj: can you rebase?

**Comment by danielqsj** [comment]
@pohly rebase has some issues. I open another PR for this fix https://github.com/kubernetes/kubernetes/pull/85190.

**Pull request closed without merging**